v1/version/version.go

```diff
@@ -31,18 +31,28 @@ func init() {
 	if !ok {
 		return
 	}
-	dirty := false
+	var dirty bool
+	var binTimestamp, binVcs string
+
 	for _, s := range bi.Settings {
 		switch s.Key {
 		case "vcs.time":
-			Timestamp = s.Value
+			binTimestamp = s.Value
 		case "vcs.revision":
-			Vcs = s.Value
+			binVcs = s.Value
 		case "vcs.modified":
 			dirty = s.Value == "true"
 		}
 	}
-	if dirty {
-		Vcs += "-dirty"
+
+	if Timestamp == "" {
+		Timestamp = binTimestamp
+	}
```

Comment:
[nit]
```go
Timestamp = cmp.Or(Timestamp, binTimestamp)
```

but this is fine, no need to change it.

Reply:
Cool I wasn't aware of `cmp.Or` ! That does look cleaner.